### ISSUE TITLE
Events `cancel` and `close` on <dialog> reported by `no-unknown-event`

### ISSUE THREAD

**Issue by jrencz** [runem/lit-analyzer#286]
Code:
```ts
class MyElementWithDialog extends LitElement {
  override render() {
    return html`<dialog
      @click="${this.onDialogClick}"
      @close="${this.onDialogClose}"
      @cancel="${this.onDialogCancel}"
    >Content</dialog>
  }
}
```

Reports

```
    Unknown event 'cancel'. Did you mean '@change'?
    6:  @cancel="${this.onDialogCancel}"
    no-unknown-event


    Unknown event 'close'.
    6:  @close="${this.onDialogClose}"
    no-unknown-event
```

Both events exist and work
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/close_event
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/cancel_event

but they don't seem to be known to the tool.

Workaround:

```ts
declare global {
  /**
   * @emits cancel
   * @emits close
   */
  interface HTMLDialogElement {}
}
```

**Comment by didnotwant** [comment]
The same error is raised for the `paste` event.
